nkms/crypto/powers.py

```diff
@@ -113,8 +126,23 @@ class EncryptingPower(KeyPairBasedPower):
     provides = ("decrypt",)
 
 
-class DelegatingPower(KeyPairBasedPower):   
-    _keypair_class = EncryptingKeypair
-    not_found_error = PowerUpError
-    provides = ("generate_kfrags",)
+class DelegatingPower(DerivedKeyBasedPower):
+
+    def __init__(self):
+        self.umbral_keying_material = UmbralKeyingMaterial()
+
+    def generate_kfrags(self, bob_pubkey_enc, label, m, n) -> Union[UmbralPublicKey, List]:
+        """
+        Generates re-encryption key frags ("KFrags") and returns them.
+
+        These KFrags can be used by Ursula to re-encrypt a Capsule for Bob so
+        that he can activate the Capsule.
+        :param bob_pubkey_enc: Bob's public key
+        :param m: Minimum number of KFrags needed to rebuild ciphertext
+        :param n: Total number of rekey shares to generate
+        """
+        # TODO: salt?
```

Comment:
Btw if we have application-specific salts, empty labels may start making sense.. But perhaps we can think of it after the testnet release